### PR TITLE
HOTFIX: follow up on KAFKA-4275

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
@dguy @guozhangwang @hjafarpour @enothereska @miguno

**Comment by guozhangwang** [comment]
@mjsax could you add a unit test since this is not exposed by the current coverage.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java [hunk 1]
```diff
@@ -301,8 +301,8 @@ public V apply(Change<V> change) {
         topology.addProcessor(joinThisName, joinThis, this.name);
         topology.addProcessor(joinOtherName, joinOther, ((KTableImpl) other).name);
         topology.addProcessor(joinMergeName, joinMerge, joinThisName, joinOtherName);
-        topology.connectProcessorAndStateStores(joinThisName, other.getStoreName());
-        topology.connectProcessorAndStateStores(joinOtherName, getStoreName());
+        topology.connectProcessorAndStateStores(joinThisName, ((KTableImpl) other).valueGetterSupplier().storeNames());
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java:304]: My bad, should have found that.

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java:304]: Glad you take the blame :P

**Comment by mjsax** [comment]
@guozhangwang Added test.

**Comment by guozhangwang** [comment]
Unit test run passed locally. LGTM.

**Comment by guozhangwang** [comment]
Merged to trunk.

**Comment by guozhangwang** [comment]
@mjsax  Could you file a follow-up PR for 0.10.1 branch just in case we have a bug-fix release later?

**Comment by mjsax** [comment]
#1992 was only merged to trunk, too. Should this really go into `0.10.1`? If yes, I can open one PR for both together.

**Pull request closed without merging**